imports/wasi_snapshot_preview1/fs.go

```diff
@@ -1432,9 +1428,23 @@ func pathFilestatGetFn(_ context.Context, mod api.Module, params []uint64) Errno
 		return errno
 	}
 
-	// Stat the file without allocating a file descriptor
+	// Stat the file without allocating a file descriptor.
+	//
+	// Note: `preopen` is a `sysfs.FS` interface, so passing the address of `st`
+	// causes the value to escape to the heap because the compiler doesn't know
+	// whether the pointer will be retained by the method.
+	//
+	// This could be optimized by modifying Stat/Lstat to return the `Stat_t`
+	// value instead of passing a pointer as output parameter.
```

Comment:
This is unrelated to the change but I wanted to record this observation somewhere, let me know if you'd rather have it somewhere else.

Reply:
interesting point, let's leave it for now. I guess we can't solve this with `//go:noescape`? In any case, it was only conventional to pass like syscall.Stat. Feel free to change in this PR or next.